bitmex-stream/src/lib.rs

```diff
@@ -16,16 +25,41 @@ pub use tokio_tungstenite::tungstenite::Error;
 pub fn subscribe<const N: usize>(
```

Comment:
in order not to break the public API (especially for users not interested in passing in credentials), it would be better to create an additional public function `subscribe_with_credentials` and then both forward to this implementation.

I know you want this sooner rather than later in production, so I'll quickly create a PR that shifts that around so that you can have a look if it's needed over the weekend.

It's probably not worth creating a builder pattern here as it's unlikely to grow more than that, but it's also an option 👯‍♂️

Reply:
feel free to cherry-pick and squash the commit: https://github.com/itchysats/itchysats/pull/2529/files

Reply:
Thanks. Next time feel free to push on top of this PR :)